components/org.wso2.carbon.appmgt.usage/org.wso2.carbon.appmgt.usage.publisher/src/main/java/org/wso2/carbon/appmgt/usage/publisher/APIMgtUsageHandler.java

```diff
@@ -356,6 +362,54 @@ public WebApp getWebApp(String context,String version) throws AppManagementExcep
           return webApp;
     }
 
+    /**
+     * Get WebApp for default version web apps.
+     * @param context
+     * @param tenantDomain
+     * @return
+     * @throws AppManagementException
+     */
+    public WebApp getNonVersionedWebApp(String context, String tenantDomain) throws AppManagementException {
+        Connection conn = null;
+        PreparedStatement ps = null;
+        ResultSet rs = null;
+        WebApp webApp = null;
+
+        String sqlQuery = "SELECT APM_APP.APP_NAME, APM_APP.APP_PROVIDER, APM_APP.TRACKING_CODE, " +
+                "APM_APP_DEFAULT_VERSION.PUBLISHED_DEFAULT_APP_VERSION FROM APM_APP LEFT JOIN " +
+                "APM_APP_DEFAULT_VERSION ON APM_APP_DEFAULT_VERSION.APP_NAME=APM_APP.APP_NAME AND " +
+                "APM_APP_DEFAULT_VERSION.APP_PROVIDER=APM_APP.APP_PROVIDER WHERE APM_APP.CONTEXT=? AND APM_APP" +
+                ".TENANT_ID=?";
+        try {
+            conn = APIMgtDBUtil.getConnection();
+            ps = conn.prepareStatement(sqlQuery);
+            int tenantId = UsageComponent.getRealmService().getTenantManager().getTenantId(tenantDomain);
+            ps.setString(1,context);
+            ps.setInt(2,tenantId);
+            rs = ps.executeQuery();
+            while(rs.next()){
```

Comment:
Fixed in https://github.com/wso2/carbon-appmgt/pull/539/commits/207d3b1c14192fee91f6d02b39c54f42bdf46e92